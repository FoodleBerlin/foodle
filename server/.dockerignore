*.png
*.md
*.jpg
#node_modules/ # Some issues appear if this one is not commented out, idk why exactly. TODO: Investigate at later on
tests/