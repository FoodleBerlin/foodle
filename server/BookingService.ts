/* import { Context } from '@apollo/client';
import { Booking, BookingStatus, Frequency } from '@prisma/client';

export function findDaySlots() {}

export function mapRequestedSlotsToExistingSlots() {}

export async function createBooking(
  ctx: Context,
  userId: string,
  propertyId: string,
  price: number,
  startDate: Date,
  endDate: Date,
  frequency: Frequency
): Booking {
  return await ctx.prisma.booking.create({
    data: {
      tenantId: userId,
      propertyId: propertyId,
      bookingStatus: BookingStatus.pending,
      totalPrice: price,
      startDate: startDate,
      endDate: endDate,
      frequency: frequency,
    },
  });
}
 */
