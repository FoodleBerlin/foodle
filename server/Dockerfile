FROM node:16
EXPOSE 5000
RUN apt-get update && apt-get install -y apt-transport-https ca-certificates curl gnupg && \
    curl -sLf --retry 3 --tlsv1.2 --proto "=https" 'https://packages.doppler.com/public/cli/gpg.DE2A7741A397C129.key' | apt-key add - && \
    echo "deb https://packages.doppler.com/public/cli/deb/debian any-version main" | tee /etc/apt/sources.list.d/doppler-cli.list && \
    apt-get update && \
    apt-get -y install doppler

#setup yarn 
RUN npm i -g corepack
RUN corepack prepare yarn@1.22.19 --activate
WORKDIR /
COPY ./ ./server
WORKDIR /server
RUN yarn
RUN yarn prisma:generate
RUN yarn tsc --build
CMD [ "doppler", "run", "--","node", "build/index.js" ]