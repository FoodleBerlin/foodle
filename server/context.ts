import prisma from "./singletons/prisma";
import { PrismaClient } from "@prisma/client";
import { Request, Response } from "express";

export type Context = {
  req: any;
  res: any;
  user: any | null;
  prisma: PrismaClient;
};

export async function createContext(
  req: Request,
  res: Response
): Promise<Context> {
  return {
    req: res,
    res: req,
    user: null, // TODO get is from req.cookies.jwt,
    prisma,
  };
}
