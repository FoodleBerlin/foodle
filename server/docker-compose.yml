version: "3.8"
services:
  # postgresql://foodle:foodle@localhost:5431
  postgres:
    image: postgres
    container_name: foodle
    restart: always
    environment:
      - POSTGRES_USER=foodle
      - POSTGRES_PASSWORD=foodle
    ports:
      - "5431:5432"
