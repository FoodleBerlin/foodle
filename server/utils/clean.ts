import prisma from '../singletons/prisma';

export async function clean() {
  try {
    await prisma.$executeRaw`DELETE FROM "User";`;
  } catch (e) {
    throw new Error(`Unable to seed ${e}`);
  }
}
