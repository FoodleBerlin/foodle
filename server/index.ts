import { ApolloServer } from 'apollo-server-express';
import schema from './schema';
import { createContext } from './context';
import express from 'express';
import session from 'express-session';
import passport from './passport';
import forgeJWT from '../utils/forgeJWT';
import { DataSources } from 'apollo-server-core/dist/graphqlOptions';
import StripeWrapper from './singletons/stripe/endpoints';
import datasources from './singletons/datasources';

const port = process.env.PORT || 5000

export const app = express();
app.use(passport.initialize());

app.use(
  session({
    secret: process.env.SERVER_SECRET ?? '',
    resave: false,
    saveUninitialized: false,
  })
);
export const apollo: ApolloServer = new ApolloServer({
  // An executable GraphQL schema.
  schema,
  // An object (or a function that creates an object) that's passed to every resolver that executes for a particular operation.
  introspection: true,
  // This enables resolvers to share helpful context, such as a database connection.
  context: createContext,
  dataSources: () => datasources() as DataSources<Record<'stripeWrapper', StripeWrapper>>,
});

export const router = express.Router();

export async function main() {
  await apollo.start();
  app.use(router);
  apollo.applyMiddleware({ app });
  app.listen({
    port: port,
  });
}
if (!process.env.TEST) {
  main();
}

router.get('/api/auth', passport.authenticate('google', { scope: ['profile', 'email'] }));

router.get('/api/callback', (req: any, res: any, next) => {
  passport.authenticate('google', async (err: any, user: any) => {
    const token = await forgeJWT(user);
    res.cookie('jwt', token, {
      httpOnly: true,
      secure: false, // true in prod,
      sameSite: 'lax', // 'strict' in prod,
    });
    return res.redirect(process.env.CLIENT_URL);
  })(req, res, next);
});
