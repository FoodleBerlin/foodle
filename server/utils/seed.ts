import prisma from '../singletons/prisma';
import { clean } from './clean';
export async function seed() {
  await clean();
  const [users] = await Promise.all(
    await prisma.$transaction([
      prisma.user.createMany({
        data: [
          {
            id: '1',
            email: 'user1@gmail.com',
            handle: 'user1',
            fullName: 'User 1',
            role: 'tenant',
            zip: 13000,
            description: 'desc',
            passportS3Id: 'passS3',
            solvencyS3Id: 'solvS3',
            licenseS3Id: 'liceS3',
          },
          {
            id: '2',
            email: 'use2@gmail.com',
            handle: 'user2',
            fullName: 'User 2',
            role: 'tenant',
            zip: 14000,
            description: 'desc',
            passportS3Id: 'passS3',
            solvencyS3Id: 'solvS3',
            licenseS3Id: 'liceS3',
          },
          {
            id: '3',
            email: 'user3@gmail.com',
            handle: 'user3',
            fullName: 'User 3',
            role: 'tenant',
            zip: 14000,
            description: 'desc',
            passportS3Id: 'passS3',
            solvencyS3Id: 'solvS3',
            licenseS3Id: 'liceS3',
          },
          {
            id: '4',
            email: 'user4@gmail.com',
            handle: 'user4',
            fullName: 'User 4',
            role: 'landlord',
            zip: 15000,
            description: 'desc',
            passportS3Id: 'passS3',
            solvencyS3Id: 'solvS3',
            licenseS3Id: 'liceS3',
          },
          {
            id: '5',
            email: 'user5@gmail.com',
            handle: 'user5',
            fullName: 'User 5',
            role: 'tenantLandlord',
            zip: 16000,
            description: 'desc',
            passportS3Id: 'passS3',
            solvencyS3Id: 'solvS3',
            licenseS3Id: 'liceS3',
          },
        ],
      }),
    ])
  );

  await prisma.$transaction([
    prisma.property.createMany({
      data: [
        {
          zip: 1355,
          size: 123,
          street: 'sample-street',
          streetNumber: '23',
          city: 'Berlin',
          description: 'user1-listing',
          serviceFee: 5,
          rules: ['rule-1', 'rule-2'],
          title: 'prop title',
          hourlyPrice: 12,
          deposit: 2,
          images: ['test-image-url'],
          partialSpace: false,
          ownerId: '1',
          handle: 'prop1',
          frequency: 'weekly',
          facilities: ['Unfurnished'],
          startDate: '2022-06-27T08:00:00.003Z',
          endDate: '2022-06-30T08:00:00.003Z',
        },
        {
          zip: 1355,
          size: 123,
          street: 'sample-street',
          streetNumber: '23',
          city: 'Berlin',
          description: 'user1-listing',
          serviceFee: 5,
          rules: ['rule-1', 'rule-2'],
          title: 'prop title',
          hourlyPrice: 12,
          deposit: 2,
          images: ['test-image-url'],
          partialSpace: false,
          ownerId: '2',
          handle: 'prop2',
          frequency: 'weekly',
          facilities: ['Elevator', 'Parking', 'Dishwasher'],
          startDate: '2022-08-27T08:00:00.003Z',
          endDate: '2022-09-30T08:00:00.003Z',
        },
        {
          zip: 1355,
          size: 123,
          street: 'sample-street',
          streetNumber: '23',
          city: 'Berlin',
          description: 'user1-listing',
          serviceFee: 5,
          rules: ['rule-1', 'rule-2'],
          title: 'prop title',
          hourlyPrice: 12,
          deposit: 2,
          images: ['test-image-url'],
          partialSpace: false,
          ownerId: '2',
          handle: 'prop3',
          frequency: 'weekly',
          facilities: ['Elevator', 'Parking'],
          startDate: '2022-10-27T08:00:00.003Z',
          endDate: '2022-11-30T08:00:00.003Z',
        },
      ],
    }),
  ]);

  const prop1 = await prisma.property.findUnique({
    where: {
      handle: 'prop1',
    },
  });
  if (prop1 == null) {
    throw Error('Error while seeding db, property with handle prop1 does not exist.');
  }
  const prop2 = await prisma.property.findUnique({
    where: {
      handle: 'prop2',
    },
  });
  if (prop2 == null) {
    throw Error('Error while seeding db, property with handle prop2 does not exist.');
  }
  // seeding for booking request
  /*  
      test weekly, one weekday
      {
        daySlots: {
          startTime: '2022-06-27T08:00:00.003Z',
          endTime: '2022-06-27T16:00:00.003Z',
        },
        frequency: 'NONE',
        startDate: '2022-06-27T08:00:00.003Z',
        endDate: '2022-07-25T08:00:00.003Z',
        propertyHandle: 'prop1',
      } */
  await prisma.$transaction([
    prisma.daySlot.createMany({
      data: [
        {
          startTime: '2022-06-27T08:00:00.003Z',
          endTime: '2022-06-27T16:00:00.003Z',
          propertyId: prop1.id,
        },
        {
          startTime: '2022-07-04T08:00:00.003Z',
          endTime: '2022-07-04T16:00:00.003Z',
          propertyId: prop1.id,
        },
        {
          startTime: '2022-07-11T08:00:00.003Z',
          endTime: '2022-07-11T16:00:00.003Z',
          propertyId: prop1.id,
        },
        {
          startTime: '2022-07-18T08:00:00.003Z',
          endTime: '2022-07-18T16:00:00.003Z',
          propertyId: prop1.id,
        },
        {
          startTime: '2022-07-25T08:00:00.003Z',
          endTime: '2022-07-25T16:00:00.003Z',
          propertyId: prop1.id,
        },

        /* 
          test monthly + multiple weekdays + varying day time
          
          const bookingVars = {
            daySlots: [
              {
                startTime: '2022-05-23T10:00:00.003Z',
                endTime: '2022-05-23T16:00:00.003Z',
              },
              {
                startTime: '2022-05-25T08:00:00.003Z',
                endTime: '2022-05-25T16:00:00.003Z',
              },
            ],
            frequency: 'MONTHLY',
            startDate: '2022-05-25T08:00:00.003Z',
            endDate: '2022-07-22T08:00:00.003Z',
            propertyHandle: 'prop2',
          };
        */

        {
          startTime: '2022-05-25T08:00:00.003Z',
          endTime: '2022-05-25T16:00:00.003Z',
          propertyId: prop2.id,
        },
        {
          startTime: '2022-05-23T08:00:00.003Z',
          endTime: '2022-05-23T16:00:00.003Z',
          propertyId: prop2.id,
        },
        {
          startTime: '2022-06-20T08:30:00.003Z',
          endTime: '2022-06-20T16:00:00.003Z',
          propertyId: prop2.id,
        },
        {
          startTime: '2022-06-22T08:00:00.003Z',
          endTime: '2022-06-22T16:00:00.003Z',
          propertyId: prop2.id,
        },
        {
          startTime: '2022-07-18T09:00:00.003Z',
          endTime: '2022-07-18T16:00:00.003Z',
          propertyId: prop2.id,
        },
        {
          startTime: '2022-07-20T08:00:00.003Z',
          endTime: '2022-07-20T16:00:00.003Z',
          propertyId: prop2.id,
        },
        /* 
          test frequency none
          {
          "daySlots": {
            "startTime": "2022-06-27T09:00:00.003Z",
            "endTime": "2022-06-27T13:00:00.003Z",
            "weekday": "FRI"
          },
           {
            "startTime": "2022-06-27T08:00:00.003Z",
            "endTime": "2022-06-27T16:00:00.003Z",
            "weekday": "WED"
          },
          "frequency": "NONE",
          "startDate": "2022-05-25T08:00:00.003Z",
          "endDate": "2022-07-22T08:00:00.003Z",
          "propertyHandle": "prop2"
        } */

        {
          startTime: '2022-08-19T08:00:00.003Z',
          endTime: '2022-08-19T16:00:00.003Z',
          propertyId: prop2.id,
        },
      ],
    }),
  ]);
}
