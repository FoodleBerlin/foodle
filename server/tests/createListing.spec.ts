import { clean } from '../utils/clean';
import { seed } from '../utils/seed';
import { apollo as server } from './server/index';

beforeAll(async () => {
  await clean();
  await seed();
});

const descriptionToLong =
  'Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor. Aenean massa. Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Donec quam felis, ultricies nec, pellentesque eu, pretium quis, sem. Nulla consequat massa quis enim. Donec pede justo, fringilla vel, aliquet nec, vulputate eget, arcu. In enim justo, rhoncus ut, imperdiet a, venenatis vitae, justo. Nullam dictum felis eu pede mollis pretium. Integer tincidunt. Cras dapibus. Vivamus elementum semper nisi. Aenean vulputate eleifend tellus. Aenean leo ligula, porttitor eu, consequat vitae, eleifend ac, enim. Aliquam lorem ante, dapibus in, viverra quis, feugiat a, tellus. Phasellus viverra nulla ut metus varius laoreet. Quisque rutrum. Aenean imperdiet. Etiam ultricies nisi vel augue. Curabitur ullamcorper ultricies nisi. Nam eget dui. Etiam rhoncus. Maecenas tempus, tellus eget condimentum rhoncus, sem quam semper libero, sit amet adipiscing sem neque sed ipsum. Nsdfsfsfsdfsdf';

const query = `
mutation CreateListing(
  $size: Int!
  $title: String!
  $ownerHandle: String!
  $street: String!
  $streetNumber: String!
  $zip: Int!
  $city: String!
  $description: String!
  $hourlyPrice: Int!
  $serviceFee: Int!
  $rules: [String!]!
  $deposit: Int!
  $images: [String!]!
  $partialSpace: Boolean!
  $startDate: DateTime!
  $endDate: DateTime!
  $frequency: FrequencyEnum!
  $facilities: [String!]!
  $availableDays: [AvailableDay!]!
  $pickup: Boolean
) {
  createListing(
    size: $size
    title: $title
    ownerHandle: $ownerHandle
    street: $street
    streetNumber: $streetNumber
    zip: $zip
    city: $city
    description: $description
    hourlyPrice: $hourlyPrice
    serviceFee: $serviceFee
    rules: $rules
    deposit: $deposit
    images: $images
    partialSpace: $partialSpace
    startDate: $startDate
    endDate: $endDate
    frequency: $frequency
    facilities: $facilities
    availableDays: $availableDays
    pickup: $pickup
  ) {
    Property {
      kind
      rules
      handle
      daySlots {
        startTime
        endTime
      }
      title
      size
      owner {
        dob
      }
      street
      streetNumber
      zip
      city
      description
      images
      partialSpace
      deposit
      pickup
      isVerified
      hourlyPrice
      serviceFee
      facilities
      frequency
    }
    ClientErrorUserNotExists {
      message
    }
    ClientErrorInvalidInput {
      message
    }
    NoAvailableSlots {
      message
    }
    UnknownError {
      message
    }
  }
}
    `;

/*
 AvailableDays are not tested, as they are retrieved from the db in a different order each time therefore
 snaps are never equal. DaySlot logic is tested separately in unit tests
  Todo: test if daySlots get saved to db correctly => user different owners
*/

const stdVars = {
  zip: 1355,
  size: 123,
  street: 'sample-street',
  streetNumber: '23',
  city: 'Berlin',
  description: 'user1-listing',
  serviceFee: 5,
  rules: ['rule-1', 'rule-2'],
  title: 'prop title',
  hourlyPrice: 12,
  deposit: 2,
  images: ['test-image-url'],
  partialSpace: false,
  ownerHandle: 'user4',
  ownerId: '1',
  handle: 'prop1',
  frequency: 'WEEKLY',
  facilities: ['Unfurnished'],
  startDate: '2022-06-27T08:00:00.003Z',
  endDate: '2022-06-30T08:00:00.003Z',
  availableDays: [
    {
      startTime: '2022-01-07T09:00:00.000Z',
      endTime: '2022-01-07T12:00:00.000Z',
    },
  ],
};

describe(' Property', () => {
  it('can create a listing, frequency weekly', async () => {
    const vars = { ...stdVars };
    vars.frequency = 'WEEKLY';
    const res = await server.executeOperation({
      query,
      variables: {
        ...vars,
      },
    });
    expect(res).toMatchSnapshot();
  });

  it('can create a listing, frequency monthly', async () => {
    const vars = { ...stdVars };
    vars.frequency = 'MONTHLY';
    const res = await server.executeOperation({
      query,
      variables: {
        ...vars,
      },
    });
    expect(res).toMatchSnapshot();
  });

  it('can create a listing, frequency none', async () => {
    const vars = { ...stdVars };
    vars.frequency = 'NONE';
    const res = await server.executeOperation({
      query,
      variables: {
        ...vars,
      },
    });
    expect(res).toMatchSnapshot();
  });

  describe(' Property', () => {
    it('can create a listing, frequency weekly and multiple days per week', async () => {
      const vars = { ...stdVars };
      vars.availableDays = [
        {
          startTime: '2022-01-01T10:00:00.000Z',
          endTime: '2022-01-01T13:00:00.000Z',
        },
        {
          startTime: '2022-01-01T08:00:00.000Z',
          endTime: '2022-01-01T14:00:00.000Z',
        },
      ];
      const res = await server.executeOperation({
        query,
        variables: {
          ...vars,
        },
      });
      expect(res).toMatchSnapshot();
    });

    it('it fails when a listing city string arg out of max range', async () => {
      const vars = { ...stdVars };
      vars.city =
        'Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor. Aenean mm';

      const res = await server.executeOperation({
        query,
        variables: {
          ...vars,
        },
      });
      expect(res).toMatchSnapshot();
    });

    it('it fails when a listing zip code arg out of max range', async () => {
      const vars = { ...stdVars };
      vars.zip = 123456;
      const res = await server.executeOperation({
        query,
        variables: {
          ...vars,
        },
      });
      expect(res).toMatchSnapshot();
    });
    it('it fails when a listing street arg out of max range', async () => {
      const vars = { ...stdVars };
      vars.street =
        'Lorem ipsum dolor sit amet, consectetsuer adipiscing elit. Aenean commodo ligula eget dolor. Aenean mm';
      const res = await server.executeOperation({
        query,
        variables: {
          ...vars,
        },
      });
      expect(res).toMatchSnapshot();
    });

    it('it fails when a listing descriptiosn arg out of max range', async () => {
      const vars = { ...stdVars };
      vars.description = descriptionToLong;
      const res = await server.executeOperation({
        query,
        variables: {
          ...vars,
        },
      });
      expect(res).toMatchSnapshot();
    });

    it('it fails when startDate is after endDate', async () => {
      const vars = { ...stdVars };
      vars.startDate = '2022-05-15T16:02:51.063Z';
      const res = await server.executeOperation({
        query,
        variables: {
          ...vars,
        },
      });
      expect(res).toMatchSnapshot();
    });

    it('it fails when availableDays is empty', async () => {
      const vars = { ...stdVars };
      vars.availableDays = [];
      const res = await server.executeOperation({
        query,
        variables: {
          ...vars,
        },
      });
      expect(res).toMatchSnapshot();
    });

    it('it fails when wrong DateTime format', async () => {
      const vars = { ...stdVars };
      vars.startDate = '2022-05-15T16:051.0';
      const res = await server.executeOperation({
        query,
        variables: {
          ...vars,
        },
      });
      expect(res).toMatchSnapshot();
    });
  });
});
