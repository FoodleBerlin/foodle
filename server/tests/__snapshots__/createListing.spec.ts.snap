// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Property  Property can create a listing, frequency weekly and multiple days per week 1`] = `
Object {
  "data": Object {
    "createListing": Object {
      "ClientErrorInvalidInput": null,
      "ClientErrorUserNotExists": null,
      "NoAvailableSlots": null,
      "Property": Object {
        "city": "Berlin",
        "daySlots": Array [
          Object {
            "endTime": 2022-07-02T12:00:00.003Z,
            "startTime": 2022-07-02T09:00:00.003Z,
          },
          Object {
            "endTime": 2022-07-02T13:00:00.003Z,
            "startTime": 2022-07-02T07:00:00.003Z,
          },
        ],
        "deposit": 2,
        "description": "user1-listing",
        "facilities": Array [
          "Unfurnished",
        ],
        "frequency": "WEEKLY",
        "handle": "prop_title_a4859a",
        "hourlyPrice": 12,
        "images": Array [
          "test-image-url",
        ],
        "isVerified": false,
        "kind": "property",
        "owner": Object {
          "dob": null,
        },
        "partialSpace": false,
        "pickup": false,
        "rules": Array [
          "rule-1",
          "rule-2",
        ],
        "serviceFee": 5,
        "size": 123,
        "street": "sample-street",
        "streetNumber": "23",
        "title": "prop title",
        "zip": 1355,
      },
      "UnknownError": null,
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[` Property  Property it fails when a listing city string arg out of max range 1`] = `
Object {
  "data": Object {
    "createListing": Object {
      "ClientErrorInvalidInput": Object {
        "message": "City name Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor. Aenean mm is invalid, must have a max length of 50 characters.",
      },
      "ClientErrorUserNotExists": null,
      "NoAvailableSlots": null,
      "Property": null,
      "UnknownError": null,
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[` Property  Property it fails when a listing descriptiosn arg out of max range 1`] = `
Object {
  "data": Object {
    "createListing": Object {
      "ClientErrorInvalidInput": Object {
        "message": "Description Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor. Aenean massa. Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Donec quam felis, ultricies nec, pellentesque eu, pretium quis, sem. Nulla consequat massa quis enim. Donec pede justo, fringilla vel, aliquet nec, vulputate eget, arcu. In enim justo, rhoncus ut, imperdiet a, venenatis vitae, justo. Nullam dictum felis eu pede mollis pretium. Integer tincidunt. Cras dapibus. Vivamus elementum semper nisi. Aenean vulputate eleifend tellus. Aenean leo ligula, porttitor eu, consequat vitae, eleifend ac, enim. Aliquam lorem ante, dapibus in, viverra quis, feugiat a, tellus. Phasellus viverra nulla ut metus varius laoreet. Quisque rutrum. Aenean imperdiet. Etiam ultricies nisi vel augue. Curabitur ullamcorper ultricies nisi. Nam eget dui. Etiam rhoncus. Maecenas tempus, tellus eget condimentum rhoncus, sem quam semper libero, sit amet adipiscing sem neque sed ipsum. Nsdfsfsfsdfsdf is invalid, must have a max length of 1000 characters.",
      },
      "ClientErrorUserNotExists": null,
      "NoAvailableSlots": null,
      "Property": null,
      "UnknownError": null,
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[` Property  Property it fails when a listing street arg out of max range 1`] = `
Object {
  "data": Object {
    "createListing": Object {
      "ClientErrorInvalidInput": Object {
        "message": "Street name Lorem ipsum dolor sit amet, consectetsuer adipiscing elit. Aenean commodo ligula eget dolor. Aenean mm is invalid, must have a max length of 50 characters.",
      },
      "ClientErrorUserNotExists": null,
      "NoAvailableSlots": null,
      "Property": null,
      "UnknownError": null,
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[` Property  Property it fails when a listing zip code arg out of max range 1`] = `
Object {
  "data": Object {
    "createListing": Object {
      "ClientErrorInvalidInput": Object {
        "message": "Zip code 123456 is invalid, must have a max length of 5 characters.",
      },
      "ClientErrorUserNotExists": null,
      "NoAvailableSlots": null,
      "Property": null,
      "UnknownError": null,
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[` Property  Property it fails when availableDays is empty 1`] = `
Object {
  "data": Object {
    "createListing": Object {
      "ClientErrorInvalidInput": Object {
        "message": "List argument availableDays must not be empty.",
      },
      "ClientErrorUserNotExists": null,
      "NoAvailableSlots": null,
      "Property": null,
      "UnknownError": null,
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[` Property  Property it fails when startDate is after endDate 1`] = `
Object {
  "data": Object {
    "createListing": Object {
      "ClientErrorInvalidInput": null,
      "ClientErrorUserNotExists": null,
      "NoAvailableSlots": null,
      "Property": Object {
        "city": "Berlin",
        "daySlots": Array [
          Object {
            "endTime": 2022-05-27T11:00:51.063Z,
            "startTime": 2022-05-27T08:00:51.063Z,
          },
          Object {
            "endTime": 2022-06-03T11:00:51.063Z,
            "startTime": 2022-06-03T08:00:51.063Z,
          },
          Object {
            "endTime": 2022-06-17T11:00:51.063Z,
            "startTime": 2022-06-17T08:00:51.063Z,
          },
          Object {
            "endTime": 2022-06-10T11:00:51.063Z,
            "startTime": 2022-06-10T08:00:51.063Z,
          },
          Object {
            "endTime": 2022-05-20T11:00:51.063Z,
            "startTime": 2022-05-20T08:00:51.063Z,
          },
          Object {
            "endTime": 2022-06-24T11:00:51.063Z,
            "startTime": 2022-06-24T08:00:51.063Z,
          },
        ],
        "deposit": 2,
        "description": "user1-listing",
        "facilities": Array [
          "Unfurnished",
        ],
        "frequency": "WEEKLY",
        "handle": "prop_title_b75cd4",
        "hourlyPrice": 12,
        "images": Array [
          "test-image-url",
        ],
        "isVerified": false,
        "kind": "property",
        "owner": Object {
          "dob": null,
        },
        "partialSpace": false,
        "pickup": false,
        "rules": Array [
          "rule-1",
          "rule-2",
        ],
        "serviceFee": 5,
        "size": 123,
        "street": "sample-street",
        "streetNumber": "23",
        "title": "prop title",
        "zip": 1355,
      },
      "UnknownError": null,
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[` Property  Property it fails when wrong DateTime format 1`] = `
Object {
  "data": undefined,
  "errors": Array [
    [UserInputError: Variable "$startDate" got invalid value "2022-05-15T16:051.0"; Expected type DateTime. DateTime cannot represent an invalid date-time-string 2022-05-15T16:051.0.],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[` Property can create a listing, frequency monthly 1`] = `
Object {
  "data": Object {
    "createListing": Object {
      "ClientErrorInvalidInput": null,
      "ClientErrorUserNotExists": null,
      "NoAvailableSlots": null,
      "Property": Object {
        "city": "Berlin",
        "daySlots": Array [
          Object {
            "endTime": 2022-07-01T11:00:00.003Z,
            "startTime": 2022-07-01T08:00:00.003Z,
          },
        ],
        "deposit": 2,
        "description": "user1-listing",
        "facilities": Array [
          "Unfurnished",
        ],
        "frequency": "MONTHLY",
        "handle": "prop_title_2cfdd8",
        "hourlyPrice": 12,
        "images": Array [
          "test-image-url",
        ],
        "isVerified": false,
        "kind": "property",
        "owner": Object {
          "dob": null,
        },
        "partialSpace": false,
        "pickup": false,
        "rules": Array [
          "rule-1",
          "rule-2",
        ],
        "serviceFee": 5,
        "size": 123,
        "street": "sample-street",
        "streetNumber": "23",
        "title": "prop title",
        "zip": 1355,
      },
      "UnknownError": null,
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[` Property can create a listing, frequency none 1`] = `
Object {
  "data": Object {
    "createListing": Object {
      "ClientErrorInvalidInput": null,
      "ClientErrorUserNotExists": null,
      "NoAvailableSlots": null,
      "Property": Object {
        "city": "Berlin",
        "daySlots": Array [
          Object {
            "endTime": 2022-07-01T11:00:00.003Z,
            "startTime": 2022-07-01T08:00:00.003Z,
          },
        ],
        "deposit": 2,
        "description": "user1-listing",
        "facilities": Array [
          "Unfurnished",
        ],
        "frequency": "NONE",
        "handle": "prop_title_62746b",
        "hourlyPrice": 12,
        "images": Array [
          "test-image-url",
        ],
        "isVerified": false,
        "kind": "property",
        "owner": Object {
          "dob": null,
        },
        "partialSpace": false,
        "pickup": false,
        "rules": Array [
          "rule-1",
          "rule-2",
        ],
        "serviceFee": 5,
        "size": 123,
        "street": "sample-street",
        "streetNumber": "23",
        "title": "prop title",
        "zip": 1355,
      },
      "UnknownError": null,
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[` Property can create a listing, frequency weekly 1`] = `
Object {
  "data": Object {
    "createListing": Object {
      "ClientErrorInvalidInput": null,
      "ClientErrorUserNotExists": null,
      "NoAvailableSlots": null,
      "Property": Object {
        "city": "Berlin",
        "daySlots": Array [
          Object {
            "endTime": 2022-07-01T11:00:00.003Z,
            "startTime": 2022-07-01T08:00:00.003Z,
          },
        ],
        "deposit": 2,
        "description": "user1-listing",
        "facilities": Array [
          "Unfurnished",
        ],
        "frequency": "WEEKLY",
        "handle": "prop_title_f22c2c",
        "hourlyPrice": 12,
        "images": Array [
          "test-image-url",
        ],
        "isVerified": false,
        "kind": "property",
        "owner": Object {
          "dob": null,
        },
        "partialSpace": false,
        "pickup": false,
        "rules": Array [
          "rule-1",
          "rule-2",
        ],
        "serviceFee": 5,
        "size": 123,
        "street": "sample-street",
        "streetNumber": "23",
        "title": "prop title",
        "zip": 1355,
      },
      "UnknownError": null,
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
